src/json.hpp.re2c

```diff
@@ -65,6 +65,12 @@ Class @ref nlohmann::basic_json is a good entry point for the documentation.
     #endif
 #endif
 
+// disable float-equal warnings on GCC/clang
+#if defined(__clang__) || defined(__GNUC__) || defined(__GNUG__)
+    #pragma GCC diagnostic push
+    #pragma GCC diagnostic ignored "-Wfloat-equal"
+#endif
+
```

Comment:
I think this part with the `#ifdef`s and `#pragma`s is really ugly... I am not 100% convinced yet to have something like this in the library...